src/librbd/AioImageRequest.cc

```diff
@@ -462,8 +462,15 @@ void AioImageFlush::send_request() {
     m_aio_comp->associate_journal_event(journal_tid);
   }
 
-  C_AioRequest *req_comp = new C_AioRequest(cct, m_aio_comp);
-  m_image_ctx.flush(req_comp);
+  if (!journaling && m_image_ctx.object_cacher != NULL)
```

Comment:
thanks for the education!


Sincerely, Yuan

On Tue, May 17, 2016 at 10:17 PM, Jason Dillaman <notifications@github.com>
wrote:

> In src/librbd/AioImageRequest.cc
> <https://github.com/ceph/ceph/pull/9093#discussion_r63529883>:
>
> > @@ -462,8 +462,15 @@ void AioImageFlush::send_request() {
> >      m_aio_comp->associate_journal_event(journal_tid);
> >    }
> >
> > -  C_AioRequest *req_comp = new C_AioRequest(cct, m_aio_comp);
> > -  m_image_ctx.flush(req_comp);
> > +  if (!journaling && m_image_ctx.object_cacher != NULL)
>
> No -- if the cache is disabled, you still might have in-flight
> writes/discards that you would need to ensure complete safely before
> finishing the flush. The ImageCtx::flush call handles both cases.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/9093/files/8c9101726827991f38094d035dcd388138278145#r63529883>
>